Gems/Atom/Feature/Common/Assets/Scripts/performance_metrics/timestamp_aggregator.py

```diff
@@ -35,40 +48,69 @@ def main(logs_dir):
         for entry in entries:
             name = entry['passName']
             time_ns = entry['timestampResultInNanoseconds']
-            pass_entry = pass_stats.get(name, {'max': 0, 'total': 0})
+            pass_entry = pass_stats.get(name, { 'totalTime': 0, 'maxTime': 0 })
 
-            pass_entry['max'] = max(time_ns, pass_entry['max'])
-            pass_entry['total'] += time_ns
+            pass_entry['maxTime'] = max(time_ns, pass_entry['maxTime'])
+            pass_entry['totalTime'] += time_ns
             pass_stats[name] = pass_entry
 
             frame_time += time_ns
-        frame_time_total += frame_time
-
-        frame_name = file.name.split('_')[0]
-        print(f'- Total time for frame {frame_name}: {ns_to_ms(frame_time)}ms')
-
-        frame_time_max = max(frame_time, frame_time_max)
-        frame_count += 1
-
-    if frame_count < 1:
-        print(f'No logs were found in {base_dir}')
-        exit(1)
+        frame_stats['totalTime'] += frame_time
+
+        frame_stats['maxTime'] = max(frame_time, frame_stats['maxTime'])
+        frame_stats['minTime'] = min(frame_time, frame_stats['minTime'])
+        frame_stats['count'] += 1
+
+    if frame_stats['count'] < 1:
+        raise FileNotFoundError(f'No frame timestamp logs were found in {benchmark_dir}')
+
+    frame_time_avg = frame_stats['totalTime'] / frame_stats['count']
+
+    frame_payload = copy.deepcopy(benchmark_metadata)
```

Comment:
Now that it's starting to get more complex, can you add comments to the different sections of this file to explain why you're doing what you're doing.

For example, I assume this would be something like
#Each payload starts with the metadata that uniquely identifies the test run, followed by the metrics that were collected

Without prior context, it's not straightforward to grok what's going on until you read the whole thing, and a few comments will make it easier to follow when reading top to bottom.